src/interpreter/mod.rs

```diff
@@ -133,34 +133,34 @@ pub enum CachedMir<'mir, 'tcx: 'mir> {
 
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
 /// Uniquely identifies a specific constant or static
-pub struct ConstantId<'tcx> {
+pub struct GlobalId<'tcx> {
     /// the def id of the constant/static or in case of promoteds, the def id of the function they belong to
     def_id: DefId,
     /// In case of statics and constants this is `Substs::empty()`, so only promoteds and associated
     /// constants actually have something useful here. We could special case statics and constants,
     /// but that would only require more branching when working with constants, and not bring any
     /// real benefits.
     substs: &'tcx Substs<'tcx>,
-    kind: ConstantKind,
+    kind: GlobalKind,
 }
 
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
-enum ConstantKind {
+enum GlobalKind {
     Promoted(mir::Promoted),
     /// Statics, constants and associated constants
     Global,
```

Comment:
cleaner would be to give promoteds their own def_id and remove them from MIR entirely